packages/@aws-cdk/aws-iam/test/test.policy-document.ts

```diff
@@ -315,6 +315,53 @@ export = {
     test.done();
   },
 
+  'Service principals': {
+    'regional service principals resolve appropriately'(test: Test) {
+      const stack = new Stack(undefined, undefined, { env: { region: 'cn-north-1' } });
+      const s = new PolicyStatement()
+        .addAction('test:Action')
+        .addServicePrincipal('codedeploy.amazonaws.com');
+
+      test.deepEqual(stack.node.resolve(s), {
+        Effect: 'Allow',
+        Action: 'test:Action',
+        Principal: { Service: 'codedeploy.cn-north-1.amazonaws.com.cn' }
+      });
+
+      test.done();
+    },
+
+    'regional service principals resolve appropriately (with user-set region)'(test: Test) {
+      const stack = new Stack(undefined, undefined, { env: { region: 'cn-northeast-1' } });
+      const s = new PolicyStatement()
+        .addAction('test:Action')
+        .addServicePrincipal('codedeploy.amazonaws.com', 'cn-north-1');
+
+      test.deepEqual(stack.node.resolve(s), {
+        Effect: 'Allow',
+        Action: 'test:Action',
+        Principal: { Service: 'codedeploy.cn-north-1.amazonaws.com.cn' }
+      });
+
+      test.done();
+    },
+
+    'obscure service principals resolve to the user-provided value'(test: Test) {
```

Comment:
Isn' that precisely what this test does?

Reply:
No, this test verifies the fallback in case there is no "fact" for this service principal. I was referring to the case where users "register" a fact and then consume it